syncer/sharding-meta/shardmeta.go

```diff
@@ -227,16 +246,25 @@ func (meta *ShardingMeta) ResolveShardingDDL() bool {
 	return false
 }
 
-// ActiveDDLFirstPos returns the first binlog position of active DDL
-func (meta *ShardingMeta) ActiveDDLFirstPos() (mysql.Position, error) {
+// ActiveDDLFirstLocation returns the first binlog position of active DDL
+func (meta *ShardingMeta) ActiveDDLFirstLocation() (binlog.Location, error) {
 	if meta.activeIdx >= len(meta.global.Items) {
-		return mysql.Position{}, terror.ErrSyncUnitDDLActiveIndexLarger.Generate(meta.activeIdx, meta.global.Items)
+		return binlog.Location{}, terror.ErrSyncUnitDDLActiveIndexLarger.Generate(meta.activeIdx, meta.global.Items)
 	}
-	return meta.global.Items[meta.activeIdx].FirstPos, nil
+
+	return meta.global.Items[meta.activeIdx].FirstLocation.Clone(), nil
 }
 
 // FlushData returns sharding meta flush SQL and args
 func (meta *ShardingMeta) FlushData(sourceID, tableID string) ([]string, [][]interface{}) {
+	// set FirstPosition and FirstGTIDSet for json marshal
+	for _, item := range meta.global.Items {
+		item.FirstPosition = item.FirstLocation.Position
+		if item.FirstLocation.GTIDSet != nil {
+			item.FirstGTIDSet = item.FirstLocation.GTIDSet.String()
+		}
+	}
```

Comment:
How about doing this conversion in `NewDDLItem`?

Reply:
looks good, update in 35248aa